python/ray/data/_internal/planner/exchange/vanilla_shuffle_scheduler.py

```diff
@@ -0,0 +1,100 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+from ray.data._internal.execution.interfaces import RefBundle
+from ray.data._internal.planner.exchange.interfaces import ExchangeScheduler
+from ray.data._internal.progress_bar import ProgressBar
+from ray.data._internal.remote_fn import cached_remote_fn
+from ray.data._internal.stats import StatsDict
+
+
+class VanillaShuffleScheduler(ExchangeScheduler):
+    """
+    The vanilla map-reduce shuffle scheduler.
+
+    Map tasks are first scheduled to generate map output blocks. After all map output
+    are generated, then reduce tasks are scheduled to combine map output blocks
+    together.
+
+    The concept here is similar to
+    "MapReduce: Simplified Data Processing on Large Clusters"
+    (https://dl.acm.org/doi/10.1145/1327452.1327492).
+    """
+
+    def execute(
+        self,
+        refs: List[RefBundle],
```

Comment:
This file is mostly copied from https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/shuffle.py#L55 , except the change for input and return value from `BlockList` to `RefBundle`.